pkg/controller/cassandradatacenter/services.go

```diff
@@ -62,56 +62,17 @@ func createOrUpdateNodesService(rctx *reconciliationRequestContext) (*corev1.Ser
 
 	opresult, err := controllerutil.CreateOrUpdate(context.TODO(), rctx.client, nodesService, func() error {
 		nodesService.Spec = corev1.ServiceSpec{
-			ClusterIP: "None",
-			Ports:     ports{cqlPort, jmxPort}.asServicePorts(),
-			Selector:  DataCenterLabels(rctx.cdc),
-		}
-
-		if err := controllerutil.SetControllerReference(rctx.cdc, nodesService, rctx.scheme); err != nil {
-			return err
-		}
-
-		return nil
-	})
-
-	if err != nil {
-		rctx.recorder.Event(
-			nodesService,
-			corev1.EventTypeWarning,
-			"FailureEvent",
-			fmt.Sprintf("Service %s failed to be created / updated ", nodesService.Name))
-		return nil, err
-	}
-
-	// Only log if something has changed
-	if opresult != controllerutil.OperationResultNone {
-		rctx.recorder.Event(
-			nodesService,
-			corev1.EventTypeNormal,
-			"SuccessEvent",
-			fmt.Sprintf("Service %s %s.", nodesService.Name, opresult))
-	}
-
-	return nodesService, err
-}
-
-func createOrUpdateSeedNodesService(rctx *reconciliationRequestContext) (*corev1.Service, error) {
-	seedNodesService := &corev1.Service{ObjectMeta: DataCenterResourceMetadata(rctx.cdc, "seeds")}
-	seedNodesService.Labels = SeedNodesLabels(rctx.cdc)
-
-	opresult, err := controllerutil.CreateOrUpdate(context.TODO(), rctx.client, seedNodesService, func() error {
-		seedNodesService.Spec = corev1.ServiceSpec{
 			ClusterIP:                "None",
-			Ports:                    internodePort.asServicePorts(),
+			Ports:                    ports{cqlPort, jmxPort, internodePort}.asServicePorts(),
 			Selector:                 DataCenterLabels(rctx.cdc),
 			PublishNotReadyAddresses: true,
 		}
 
-		seedNodesService.Annotations = map[string]string{
+		nodesService.Annotations = map[string]string{
```

Comment:
The (subtle) difference between the SeedNodesService and NodesService was that the former will resolve unready nodes, where as the latter will only resolve nodes that are up (responding to CQL requests).

By combining the two and always having it set `tolerate-unready-endpoints` to true means that apps can discover unready nodes.

Reply:
@zegelin why would you want to wait for unready endpoints anyway?